### DISCUSSION TITLE
Configure Renovate PR doesn't update description after renovate.json changes

### DISCUSSION THREAD

**Discussion by SantoJambit** [renovatebot/renovate#22069] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I'd like to see the "what to expect" section to reflect the renovate.json in the "Configure Renovate" PR.

### What you tried so far.

- Pushing new commits (since github had issues yesterday, I thought something maybe didn't trigger correctly)
- Waiting for a full day
- Searching for various search terms both on google and on github discussions and issues (only found a [very old one](https://github.com/renovatebot/renovate/issues/1159), which was related to package.json though)

### Relevant debug logs

I don't see any logs

**Comment by SantoJambit** [comment]
I did not open this as an issue, as I'm not sure if this is expected behavior or if I can trigger this manually somehow.

**Reply by rarkins** [reply]
Do you see any recent jobs on app.renovatebot.com?

If you see at least one job, but not any recent ones, let me know the repo name or a job ID and I can trigger a run manually. Otherwise runs should be triggered any changes to `renovate.json`

**Reply by SantoJambit** [reply]
There are jobs, the latest one showing result "disabled-closed-onboarding". Id "1069436784".

Here is the log:
```
DEBUG: No dangling containers to remove
INFO: Repository started
{
  "renovateVersion": "35.75.0"
}
DEBUG: Using localDir: /mnt/renovate/gh/my-org/my-repo
DEBUG: PackageFiles.clear() - Package files deleted
DEBUG: initRepo("my-org/my-repo")
DEBUG: Using queue: host=api.github.com, concurrency=10
DEBUG: my-org/my-repo default branch = develop
DEBUG: Using app token for git init
DEBUG: Repository cache is restored from revision 13
DEBUG: Resetting npmrc
DEBUG: checkOnboarding()
DEBUG: isOnboarded()
DEBUG: findPr(renovate/configure, Configure Renovate, !open)
DEBUG: getPrList success
{
  "pullsTotal": 3,
  "requestsTotal": 144,
  "apiQuotaAffected": true
}
DEBUG: Found PR #1013
DEBUG: Checking cached config file name
DEBUG: GET https://api.github.com/repos/my-org/my-repo/contents/renovate.json = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=404 retryCount=0, duration=166)
DEBUG: Existing config file no longer exists
DEBUG: findFile(renovate.json)
DEBUG: Initializing git repository into /mnt/renovate/gh/my-org/my-repo
DEBUG: Performing blobless clone
DEBUG: git clone completed
{
  "durationMs": 3153
}
DEBUG: latest repository commit
{
  "latestCommit": {
    "date": "2023-05-10T17:31:38+02:00",
    ...
  }
}
DEBUG: findFile(renovate.json5)
DEBUG: findFile(.github/renovate.json)
DEBUG: findFile(.github/renovate.json5)
DEBUG: findFile(.gitlab/renovate.json)
DEBUG: findFile(.gitlab/renovate.json5)
DEBUG: findFile(.renovaterc)
DEBUG: findFile(.renovaterc.json)
DEBUG: findFile(.renovaterc.json5)
DEBUG: config file not found
DEBUG: Found closed onboarding PR
DEBUG: Repo is not onboarded and no merged PRs exist
DEBUG: Getting comments for #1013
DEBUG: Found 2 comments
DEBUG: Ensuring comment "Renovate is disabled" in #1013
DEBUG: Comment is already update-to-date
INFO: Repository is disabled - skipping
DEBUG: Repository result: disabled-closed-onboarding, status: disabled, enabled: false, onboarded: undefined
DEBUG: Repository timing splits (milliseconds)
{
  "splits": {},
  "total": 128646
}
DEBUG: Package cache statistics
{
  "get": {
    "count": 0
  },
  "set": {
    "count": 0
  }
}
DEBUG: http statistics
{
  "urls": {
    "https://api.github.com/repos/my-org/my-repo/contents/renovate.json (GET,404)": 1,
    "https://api.github.com/repos/my-org/my-repo/issues/1013/comments (GET,200)": 1,
    "https://api.github.com/repos/my-org/my-repo/pulls (GET,200)": 144
  },
  "hostStats": {
    "api.github.com": {
      "requestCount": 146,
      "requestAvgMs": 832,
      "queueAvgMs": 0
    }
  },
  "totalRequests": 146
}
DEBUG: Package lookup durations
DEBUG: dns cache
{
  "hosts": [
    "api.github.com"
  ]
}
INFO: Repository finished
{
  "cloned": true,
  "durationMs": 128646
}
```

The first job from 2 days ago had "repository-changed" as a result and the second one "null". After that, each run had "disabled-closed-onboarding" as a result.

Thanks for taking a look!

**Comment by rarkins** [comment]
According to that log, 
- No config file exists in the main branch of the repo
- Onboarding PR `#1013` in the repo is no longer open

Your original question implies that the onboarding PR is still open though. Can you clarify?

**Reply by SantoJambit** [reply]
It seems we had an onboarding PR over a year ago, but it was closed due to lack of time (or maybe in favor of dependabot, I don't know for certain).

I was not aware of the old PR. I just added the repository to the renovate app 2 days ago and renovate automatically created the new PR `#2856`.

So in short: `#1013` was created in february 2022 and closed in march 2022, while `#2856` was created 2 days ago and is still open.

**Reply by SantoJambit** [reply]
The old PR had a comment from renovate, mentioning to rename the old PR to trigger a new PR. I've just done that and will report back if it fixes the issue.

**Reply by rarkins** [reply]
That's an interesting edge case. Can you let me know a job ID so I can locate the repo and look at logs on the backend?

**Reply by SantoJambit** [reply]
It gets even weirder.. there was even an onboarding PR `#1`, which got detected once I renamed `#1013`. So somehow the routine preferred the second of 3 onboarding PRs. I renamed `#1` as well and checked for other onboarding PRs. No others exist. 

Here are all the Job ids:

Job ID | Result | Date
-- | -- | --
1069917783 | null | 3 minutes ago
1069437468 | disabled-closed-onboarding | 14 hours ago
1069436784 | disabled-closed-onboarding | 15 hours ago
1068464245 | disabled-closed-onboarding | 2 days ago
1068463285 | disabled-closed-onboarding | 2 days ago
1068462865 | disabled-closed-onboarding | 2 days ago
1068393651 | disabled-closed-onboarding | 2 days ago
1068393341 | null | 2 days ago
1067837743 | repository-changed | 3 days ago

The latest Job shows null again and still no up-to-date PR description.

The log shows 404s for packages, which should be taken from the internal registry, but are not. I configured hostRules according to this: https://docs.renovatebot.com/getting-started/private-packages/. I will check that config again.

**Reply by SantoJambit** [reply, marked as answer]
So the final issue was: We had a missing config.npmrc  (since we do not commit the .npmrc file).

So in the end, the solution was:
- Rename all pre-existing onboarding PRs
- Make sure the private registries are configured correctly, so that the job doesn't fail.

Thanks for the support! If you need anything else to investigate further, let me know.

**Reply by rarkins** [reply]
OK thanks for the update. I'm pretty busy right now so I'm going to _not_ look into those job IDs unless you encounter any more problems
